### ISSUE TITLE
agent6 error: tag6 function references unknown variable $tag_name

### ISSUE THREAD

**Issue by nehalrp** [DataDog/puppet-datadog-agent#410]
Running into:
```
Error during compilation: Evaluation Error: Error while evaluating a Resource Statement, Evaluation Error: Unknown variable: 'tag_name'. at /Users/nehalrp/workspace/puppet-base/modules/components/vendor/datadog_agent/functions/tag6.pp:20:14
```

Because the function `datadog_agent::tag6` in `2.0.1` uses `$tag_name` instead of the parameter `$tag_names` (plural) in two places:
```
# For compatibility with puppet 4.6 return type is left
# out of the function signature, but `tag6` should return
# an Array

function datadog_agent::tag6(
  Variant[Array, String] $tag_names,
  Variant[String, Boolean] $lookup_fact = false,
) {
  if is_array($tag_names) {
    $tags = $tag_names.reduce([]) |$_tags , $tag| {
        concat($_tags, datadog_agent::tag6($tag, lookup_fact))
    }
  } elsif str2bool($lookup_fact) {
    $value = getvar($tag_names)

    if is_array($value){
      $tags = prefix($value, "${tag_name}:") # <<<<<<<<<< issue
    }
  } else {
    $tags = [$tag_name] # <<<<<<<<<< issue
  }

  $tags
}
```

**Comment by truthbk** [comment]
This should've been fixed in `2.1.0` can you please check @nehalrp?

**Comment by nehalrp** [comment]
It is.
